### PR TITLE
Version checking

### PR DESCRIPTION
**Pull request by BillThePlatypus**
rosflight_io checks the firmware major and minor version, and gives a warning if it does not match the rosflight version, as determined by git.

### CODE REVIEW

File: rosflight/src/rosflight_io.cpp [hunk 1]
```diff
@@ -707,8 +712,28 @@ void rosflightIO::handle_version_msg(const mavlink_message_t &msg)
     version_pub_ = nh_.advertise<std_msgs::String>("version", 1, true);
   }
   version_pub_.publish(version_msg);
+#ifdef GIT_VERSION_STRING // Macro so that is compiles even if git is not available
+  std::string git_version_string = GIT_VERSION_STRING;
```
Comment (superjax) [inline rosflight/src/rosflight_io.cpp:716]: const

File: rosflight/src/rosflight_io.cpp [hunk 2]
```diff
@@ -707,8 +712,28 @@ void rosflightIO::handle_version_msg(const mavlink_message_t &msg)
...
+  std::string rosflight_major_minor_version =
+      git_version_string.substr(0, git_version_string.find('.', git_version_string.find('.') + 1));
+  std::string firmware_version(version.version);
+  std::string firmware_major_minor_version =
+      firmware_version.substr(1, firmware_version.find('.', firmware_version.find('.') + 1) - 1);
```
Comment (superjax) [inline rosflight/src/rosflight_io.cpp:721]: is there any way we could abstract this?  (It's a little hard to grok this as-is)
```
int get_major_version(const std::string& version);
int get_minor_version(const std::string& version);
```
or whatever

File: rosflight/CMakeLists.txt [hunk 3]
```diff
@@ -32,6 +32,15 @@ if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/include/rosflight/mavlink/v1.0/.git")
   )
 endif()
 
+execute_process(COMMAND git describe --tags --abbrev=8 --always --dirty --long WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} OUTPUT_VARIABLE GIT_VERSION_STRING RESULT_VARIABLE GIT_RESULT)
+if(GIT_RESULT EQUAL 0)
+	string(REGEX REPLACE "\n$" "" GIT_VERSION_STRING "${GIT_VERSION_STRING}")
```
Comment (superjax) [inline rosflight/CMakeLists.txt:37]: why is this here again?  removing newline characters?

Comment (BillThePlatypus) [reply rosflight/CMakeLists.txt:37]: Yes, I've added a comment.
